src/samples/Teams/TeamsAgent/MyAgent.cs

```diff
@@ -0,0 +1,136 @@
+﻿using AdaptiveCards.Rendering;
+using Microsoft.Agents.Builder;
+using Microsoft.Agents.Builder.App;
+using Microsoft.Agents.Builder.App.AdaptiveCards;
+using Microsoft.Agents.Builder.State;
+using Microsoft.Agents.Core.Models;
+using Microsoft.Agents.Core.Serialization;
+using Microsoft.Agents.Extensions.Teams.App;
+using Microsoft.Agents.Extensions.Teams.Connector;
+using Microsoft.Agents.Extensions.Teams.Models;
+using System.Text.Json;
+using System.Text.Json.Nodes;
+
+namespace TeamsAgent
+{
+    public class MyAgent : AgentApplication
+    {
+        public MyAgent(AgentApplicationOptions options) : base(options)
+        {
+            RegisterExtension(new TeamsAgentExtension(this), tae =>
+            {
+                tae.OnMessageEdit(MessageEdited);
+                tae.MessageExtensions.OnSelectItem(OnSelectItem);
+                tae.MessageExtensions.OnQuery("findNuGetPackage", OnQuery);
+            });
+            OnMessageReactionsAdded(OnMessageReaction);
+            OnConversationUpdate(ConversationUpdateEvents.MembersAdded, WelcomeMessageAsync);
+            OnActivity(ActivityTypes.Message, OnMessageAsync);
+        }
+
+        private async Task<MessagingExtensionResult> OnSelectItem(ITurnContext turnContext, ITurnState turnState, object item, CancellationToken cancellationToken)
+        {
+            string jsonItem = JsonSerializer.Serialize(item);
+            await turnContext.SendActivityAsync("select item " + jsonItem, cancellationToken: cancellationToken);
+            return new MessagingExtensionResult() { ActivityPreview = new Activity() { Type = ActivityTypes.Message, Text = jsonItem } };
+        }
+
+        private async Task<MessagingExtensionResult> OnQuery(ITurnContext turnContext, ITurnState turnState, Query<IDictionary<string, object>> query, CancellationToken cancellationToken)
+        {
+
+            JsonElement el = query.Parameters.TryGetValue<JsonElement>("NuGetPackageName");

```

Comment:
@goldenwitch this line took me a while, since it's a change from the previous Query object, where we access the parameters by index. Is this expected ?